HDFDataset.py

```diff
@@ -976,14 +980,18 @@ def dump_from_dataset(self, dataset, epoch=1, start_seq=0, end_seq=float("inf"),
     else:
       default_data_input_key = data_input_keys[0]
     print("Using input data key:", default_data_input_key)
-    if len(data_target_keys) > 1:
-      if "classes" in data_target_keys:
-        default_data_target_key = "classes"
+    if len(data_target_keys) > 0:
+      if len(data_target_keys) > 1:
+        if "classes" in data_target_keys:
+          default_data_target_key = "classes"
+        else:
+          raise Exception("not sure which target data key to use from %r" % (data_target_keys,))
       else:
-        raise Exception("not sure which target data key to use from %r" % (data_target_keys,))
+        default_data_target_key = data_target_keys[0]
+      print("Using target data key:", default_data_target_key)
     else:
-      default_data_target_key = data_target_keys[0]
-    print("Using target data key:", default_data_target_key)
+      default_data_target_key = default_data_input_key
```

Comment:
If you do that, maybe you also should add it to `data_target_keys`? Not sure if this is relevant.